repository.go

```diff
@@ -1473,6 +1461,22 @@ func (r *Repository) Worktree() (*Worktree, error) {
 	return &Worktree{r: r, Filesystem: r.wt}, nil
 }
 
+func expand_ref(s storer.ReferenceStorer, ref plumbing.ReferenceName) (*plumbing.Reference, error) {
+	// return the error from the first rule
```

Comment:
Maybe something around the lines of:
```suggestion
	// For improving troubleshooting, this preserves the error for the provided `ref`, and returns the error for that specific ref in case all parse rules fails.
```

Reply:
Done